src/services/TestDataService.ts

```diff
@@ -1255,13 +1362,14 @@ export class TestDataService {
                 }
             }
         });
-        const orderDelivery = this.getBasicOrderDeliveryStruct(deliveryState, shippingMethod, customerAddress)
-        let shippingCosts = 0;
-        if (orderDelivery.shippingCosts != null) {
-            shippingCosts = orderDelivery.shippingCosts.totalPrice;
-        }
+        const orderDelivery = this.getBasicOrderDeliveryStruct(deliveryState, shippingMethod, customerAddress);
+        orderDelivery.shippingCosts ? shippingCosts = orderDelivery.shippingCosts.totalPrice : shippingCosts = 0;
+
         totalPrice += shippingCosts;
 
+        const orderTransaction = this.getBasicOrderTransactionStruct(paymentMethod, transactionState);
+        orderTransaction.amount ? orderTransaction.amount.totalPrice = totalPrice : totalPrice;
```

Comment:
@Phil23 is that possible to define somehow that `orderTransactions.amount` could not be undefined?